core/src/main/java/io/crate/types/CollectionType.java

```diff
@@ -119,13 +125,15 @@ public boolean equals(Object o) {
         if (!super.equals(o)) return false;
 
         CollectionType collectionType = (CollectionType) o;
-        return innerType.equals(collectionType.innerType);
+        return innerType != null && innerType.equals(collectionType.innerType);
```

Comment:
Isn't `innerType` now non-nullable again?

Reply:
It shouldn't be null, added a Precondition in the constructor and removing the check.